src/Polly.Core/Hedging/Controller/HedgingExecutionContext.cs

```diff
@@ -215,18 +215,18 @@ private void UpdateOriginalContext()
             }
         }
 
-        if (accepted != 1)
-        {
-            throw new InvalidOperationException($"There must be exactly one accepted outcome for hedging. Found {accepted}.");
-        }
+        Debug.Assert(accepted == 1, $"There must be exactly one accepted outcome for hedging. Found {accepted}.");
 
-        originalContext.Properties.Replace(acceptedExecution!.Properties);
-
-        if (acceptedExecution.Type == HedgedTaskType.Secondary)
+        if (acceptedExecution is not null)
         {
-            foreach (var @event in acceptedExecution.Context.ResilienceEvents)
+            originalContext.Properties.Replace(acceptedExecution!.Properties);
```

Comment:
As there's a null check we should be able to remove this?

```suggestion
            originalContext.Properties.Replace(acceptedExecution.Properties);
```

Reply:
True, not needed anymore.